File: 2026-1/sesion-de-clase-03/dotnet/seguro-vehicular/Seguros/SeguroBasico.cs
namespace Seguros;

public class SeguroBasico : ICotizable
{
    protected readonly Vehiculo vehiculo;
    public SeguroBasico(Vehiculo vehiculo)
    {
        this.vehiculo = vehiculo;
    }

    public virtual double CalcularCosto()
    {
        double costo = 500.00; // Costo Base

        if (this.vehiculo.Antiguedad > 10)
        {
            costo += 200.00;
        }
        else if (this.vehiculo.Antiguedad > 5)
        {
            costo += 100.00;
        }

        if (this.vehiculo.Marca == Marca.BMW ||
            this.vehiculo.Marca == Marca.Audi ||
            this.vehiculo.Marca == Marca.Mercedez)
        {
            costo += 100.00;
        }

        return costo;
    }

    public Vehiculo Vehiculo()
    {
        return this.vehiculo;
    }

    public virtual string Descripcion()
    {
        return "Seguro basico vehicular";
    }

    public override string ToString()
    {
        return $"Precio: {this.CalcularCosto():F2}, Descripción: {this.Descripcion()}";
    }
}

File: 2026-1/sesion-de-clase-04/java/seguro-vehicular/pe/edu/softseguros/seguros/SeguroBasico.java
package pe.edu.softseguros.seguros;

import pe.edu.softseguros.modelo.Cotizable;
import pe.edu.softseguros.modelo.Marca;
import pe.edu.softseguros.modelo.Vehiculo;

public class SeguroBasico implements Cotizable {

    protected Vehiculo vehiculo;

    public SeguroBasico(Vehiculo vehiculo) {
        this.vehiculo = vehiculo;
    }

    @Override
    public double CalcularCosto() {
        double costo = 500.00; // Costo Base

        if (this.vehiculo.getAntiguedad() > 10) {
            costo += 200.00;
        } else if (this.vehiculo.getAntiguedad() > 5) {
            costo += 100.00;
        }

        if (this.vehiculo.getMarca() == Marca.BMW
                || this.vehiculo.getMarca() == Marca.Audi
                || this.vehiculo.getMarca() == Marca.Mercedez) {
            costo += 100.00;
        }

        return costo;
    }

    @Override
    public Vehiculo Vehiculo() {
        return this.vehiculo;
    }

    @Override
    public String Descripcion() {
        return "Seguro basico vehicular";
    }

    @Override
    public String toString() {
        return String.format("Precio: %.2f, Descripción: %s",
                this.CalcularCosto(), this.Descripcion());
    }
}
